### ISSUE TITLE
Gem update requests being sent to incorrect gem host

### ISSUE THREAD

**Issue by richiethomas** [dependabot/dependabot-core#1660]
In our app's Dependabot [logs](https://app.dependabot.com/accounts/thoughtbot/update-logs/21860406), I see 401 Unauthorized errors happening for certain gems.  It looks like the latest gem versions are all being fetched from RubyGems (as expected), but any gems which require an update will trigger a request to an incorrect source host ("https://gems.graphql.pro/").  This is a host that we've configured for one gem in particular in our Gemfile, not (as far as I can tell) for all our gems.

**Comment by feelepxyz** [comment]
@richiethomas have you set up credentials in Dependabot for the private host? You can add them in the [dashboard](https://app.dependabot.com/) under your account nav > config variables.

The reason all updates try to resolve the host is because Dependabot checks if all gems in your Gemfile is resolvable. This makes sure the update hasn't broken anything.

**Comment by richiethomas** [comment]
Thank you, that helped.

I noticed the credentials were not in the dashboard, so I added them under "Config Variables" with the type of "RubyGems server".  I confirmed that the host directory where the `.gem` files are located is https://gems.graphql.pro/gems/.  I added `gems.graphql.pro/gems` as the value for the "Gem server" field and the correct username and password in the Credentials field (in the form of `username:password`, as per the instructions below the field).  The result was multiple successful dependency PRs, and a successful `Status: processed 15 minutes ago` result message.  However, I'm also seeing 404 errors in [the most recent logs](https://app.dependabot.com/accounts/thoughtbot/update-logs/22029254).  Not sure if this is a problem or not.

I'm also seeing what appears to be an incorrect host + path being used in conjunction with the requests which are 404'ing.  The error + path I see is `404 https://gems.graphql.pro:443/versions`, which is obviously using the `/versions` path instead of the `/gems` path I specified for the host in the dashboard.

This makes me wonder where the `/versions` path is coming from, since it doesn't exist in my codebase.  The gem is referenced in my Gemfile as `gem "graphql-pro", source: "https://gems.graphql.pro/"`.  Should I remove the `source` attribute and rely on the one I provided in the Dependabot dashboard instead?
